### PR TITLE
Remove feature_dehydration

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] I have licensed the changes to Element by completing the [Contributor License Agreement (CLA)](https://cla-assistant.io/element-hq/matrix-react-sdk)

Task https://github.com/element-hq/element-web/issues/26922
https://github.com/matrix-org/matrix-spec-proposals/pull/2697 is abandoned
The old dehydration feature is deprecated and not implemented in the rust crypto.
Remove the lab doc https://github.com/element-hq/element-web/pull/28173

### CODE REVIEW

**Review by dbkr — APPROVED** [review]

File: src/SecurityManager.ts [hunk 1]
```diff
@@ -278,14 +277,7 @@ async function doAccessSecretStorage(func: () => Promise<void>, forceReset: bool
...
             const keyId = Object.keys(secretStorageKeys)[0];
-            if (keyId && SettingsStore.getValue("feature_dehydration")) {
-                let dehydrationKeyInfo = {};
-                if (secretStorageKeyInfo[keyId] && secretStorageKeyInfo[keyId].passphrase) {
-                    dehydrationKeyInfo = { passphrase: secretStorageKeyInfo[keyId].passphrase };
-                }
-                logger.log("accessSecretStorage: Setting dehydration key");
-                await cli.setDehydrationKey(secretStorageKeys[keyId], dehydrationKeyInfo, "Backup device");
-            } else if (!keyId) {
+            if (!keyId) {
```
Comment (uhoreg) [inline src/SecurityManager.ts:280]: This whole `if` statement is related to dehydration, so can be removed completely -- the remaining branches are just logging, explaining why it isn't setting a decryption key

File: src/Lifecycle.ts
Comment (uhoreg) [inline src/Lifecycle.ts:1]: The `hydrateSession` function also needs to be removed (along with the two places where it's called).

File: src/SecurityManager.ts (hunk 1 above)
Comment (uhoreg) [reply src/SecurityManager.ts:280]: which means `keyId` can also be removed, since it isn't being used for anything else

**Review by uhoreg — CHANGES_REQUESTED** [review]

File: src/components/structures/auth/SoftLogout.tsx [hunk 2]
```diff
@@ -195,27 +189,15 @@ export default class SoftLogout extends React.Component<IProps, IState> {
             device_id: MatrixClientPeg.safeGet().getDeviceId() ?? undefined,
         };
 
-        let credentials: IMatrixClientCreds;
         try {
-            credentials = await sendLoginRequest(hsUrl, isUrl, loginType, loginParams);
+            await sendLoginRequest(hsUrl, isUrl, loginType, loginParams);
+            this.props.onTokenLoginCompleted();
```
Comment (florianduros) [inline src/components/structures/auth/SoftLogout.tsx:194]: `onTokenLoginCompleted` is not optional in the type declaration.

**Review by uhoreg — APPROVED** [review]

**Pull request merged**